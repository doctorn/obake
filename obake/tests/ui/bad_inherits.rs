#![allow(unused_macros)]

#[obake::versioned]
#[obake(version("0.1.0"))]
struct Foo {}

#[obake::versioned]
#[obake(version("0.1.0"))]
struct Bar {
    #[obake(inherit)]
    field_0: [Foo; 3],
}

#[obake::versioned]
#[obake(version("0.1.0"))]
enum Baz {
    Variant {
        #[obake(inherit)]
        field_0: [Foo; 3]
    }
}

fn main() {}
