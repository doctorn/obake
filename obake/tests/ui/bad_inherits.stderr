error: `#[obake(inherit)]` can only be applied to fields with `#[obake::versioned]` types
  --> $DIR/bad_inherits.rs:10:13
   |
10 |     #[obake(inherit)]
   |             ^^^^^^^

error: `#[obake(inherit)]` can only be applied to fields with `#[obake::versioned]` types
  --> $DIR/bad_inherits.rs:18:17
   |
18 |         #[obake(inherit)]
   |                 ^^^^^^^
